sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/subquery.scala

```diff
@@ -267,6 +268,33 @@ object ScalarSubquery {
   }
 }
 
+/**
+ * A subquery that is capable to return multiple scalar values.
+ */
+case class MultiScalarSubquery(
```

Comment:
Do we have to create a new subquery expression? It seems like we can just use `CreateNamedStruct` in `ScalarSubquery.plan`

Reply:
I think you are right.

Reply:
Dropped 'MultiScalarSubquery` in https://github.com/apache/spark/pull/32298/commits/1f2f75c9c52f7494db4e3bb15368c87d9b8e17c7, will change the docs and the PR description soon.